packages/documentation/copy/en/modules-reference/Theory.md

```diff
@@ -0,0 +1,415 @@
+---
+title: Theory
+layout: docs
+permalink: /docs/handbook/modules/theory.html
+oneline: How TypeScript models JavaScript modules
+translatable: true
+---
+
+## Scripts and modules in JavaScript
+
+In the early days of JavaScript, when the language only ran in browsers, there were no modules, but it was still possible to split the JavaScript for a web page into multiple files by using multiple `script` tags in HTML:
+
+```html
+<html>
+  <head>
+    <script src="a.js"></script>
+    <script src="b.js"></script>
+  </head>
+  <body></body>
+</html>
+```
+
+This approach had some downsides, especially as web pages grew larger and more complex. In particular, all scripts loaded onto the same page share the same scope—appropriately called the “global scope”—meaning the scripts had to be very careful not to overwrite each others’ variables and functions.
+
+Any system that solves this problem by giving files their own scope while still providing a way to make bits of code available to other files can be called a “module system.” (It may sound obvious to say that each file in a module system is called a “module,” but the term is often used to contrast with _script_ files, which run outside a module system, in a global scope.)
```

Comment:
The opening line for this paragraph feels a little unwieldy (lots of abstract terms) when you try say it out loud could try starting with the term and then building backwards?

```suggestion
We call a system which tries to break up that single global scope a “module system“, these are usually build around providing a per-file scope which allows for ways to pass bits of code between each other. (It may sound obvious to say that each file in a module system is called a “module,” but the term is often used to contrast with _script_ files, which run outside a module system, in a global scope.)
```